include/OpenSpaceToolkit/Physics/Environment/Atmospheric/Earth/NRLMSISE00.hpp

```diff
@@ -0,0 +1,155 @@
+/// Apache License 2.0
+
+#ifndef __OpenSpaceToolkit_Physics_Environment_Atmospheric_Earth_NRLMSISE00__
+#define __OpenSpaceToolkit_Physics_Environment_Atmospheric_Earth_NRLMSISE00__
+
+#include <OpenSpaceToolkit/Core/Containers/Array.hpp>
+#include <OpenSpaceToolkit/Core/Containers/Tuple.hpp>
+#include <OpenSpaceToolkit/Core/Types/Integer.hpp>
+#include <OpenSpaceToolkit/Core/Types/Real.hpp>
+#include <OpenSpaceToolkit/Core/Types/String.hpp>
+
+#include <OpenSpaceToolkit/Physics/Coordinate/Spherical/LLA.hpp>
+#include <OpenSpaceToolkit/Physics/Environment/Atmospheric/Model.hpp>
+#include <OpenSpaceToolkit/Physics/Time/Instant.hpp>
+#include <OpenSpaceToolkit/Physics/Units/Length.hpp>
+
+namespace ostk
+{
+namespace physics
+{
+namespace environment
+{
+namespace atmospheric
+{
+namespace earth
+{
+
+using ostk::core::types::Integer;
+using ostk::core::types::Real;
+using ostk::core::types::String;
+using ostk::core::ctnr::Tuple;
+using ostk::core::ctnr::Array;
+
+using ostk::physics::time::Instant;
+using ostk::physics::units::Length;
+using ostk::physics::coord::spherical::LLA;
+using ostk::physics::environment::atmospheric::Model;
+
+/// @brief                      NRLMSISE00 atmospheric model
+
+class NRLMSISE00 : public Model
+{
+   public:
+    /// @brief              Constructor
+
+    NRLMSISE00();
+
+    /// @brief              Clone the exponential atmospheric model
+    ///
+    /// @return             Pointer to exponential atmospheric model
+
+    virtual NRLMSISE00* clone() const override;
+
+    /// @brief              Check if the exponential atmospheric model is defined
+    ///
+    /// @return             True if the exponential atmospheric model is defined
+
+    bool isDefined() const;
+
+    /// @brief              Get the atmospheric density value at a given position and instant.
+    ///                     Optionally use provided sun position to calculate local solar time.
+    ///
+    /// @param              [in] aLLA A position, expressed as latitude, longitude, altitude [deg, deg, m]
+    /// @param              [in] anInstant An instant
+    /// @param              [in] aSunPosition Position of the sun
+    /// @return             Atmospheric density value [kg.m^-3]
+
+    Real getDensityAt(const LLA& aLLA, const Instant& anInstant, const Position& aSunPosition = Position::Undefined())
```

Comment:
Implemented a version of `getDensityAt` that takes in a sun position in order to precisely calculate the _local apparent solar time_. The NRLMSISE [model documentation](https://github.com/magnific0/nrlmsise-00/blob/master/nrlmsise-00.h#L103) actually recommends to just use `lst=sec/3600 + g_long/15` to keep variables consistent, but Orekit does it this way so I implemented it as an option to have a better comparison.

Reply:
🙌

Reply:
I should add that this is not exposed in `Earth.cpp` right now. Wanted to avoid the complexity of changing that as well.